data-exporter/src/test/java/org/reactome/release/dataexport/UniProtReactomeEntryTest.java

```diff
@@ -43,33 +54,63 @@ public void differentUniProtObjectsWithDifferentValuesNotEqual() {
 		final String DIFFERENT_DISPLAY_NAME = "UniProt:" + DIFFERENT_ACCESSION;
 
 		assertThat(
-			uniProtReactomeEntry,
-			is(not(equalTo(UniProtReactomeEntry.get(4L, DIFFERENT_ACCESSION, DIFFERENT_DISPLAY_NAME))))
+			canonicalUniProtReactomeEntry,
+			is(not(equalTo(
+				UniProtReactomeEntry.get(getNextUniProtDBID(), DIFFERENT_ACCESSION, DIFFERENT_DISPLAY_NAME))
+			))
 		);
 	}
 
 	@Test
 	public void sortsByUniProtAccessionAscendingly() {
-		final String ACCESSION_THAT_SHOULD_BE_FIRST = "A01234";
+		final String ACCESSION_THAT_SHOULD_BE_FIRST = "A0A234";
 
 		List<UniProtReactomeEntry> uniProtReactomeEntries = new ArrayList<>();
-		uniProtReactomeEntries.add(uniProtReactomeEntry);
+		uniProtReactomeEntries.add(canonicalUniProtReactomeEntry);
 		UniProtReactomeEntry uniProtReactomeEntry2 = UniProtReactomeEntry.get(
-			5L,
+			getNextUniProtDBID(),
 			ACCESSION_THAT_SHOULD_BE_FIRST,
 			DUMMY_UNIPROT_DISPLAY_NAME
 		);
 		uniProtReactomeEntries.add(uniProtReactomeEntry2);
 
 		Collections.sort(uniProtReactomeEntries);
 
-		assertThat(uniProtReactomeEntries, contains(uniProtReactomeEntry2, uniProtReactomeEntry));
+		assertThat(uniProtReactomeEntries, contains(uniProtReactomeEntry2, canonicalUniProtReactomeEntry));
+	}
+
+	@Test
+	public void isoformAccessionIsAccepted() {
+		final String ISOFORM_ACCESSION = "P12345-1";
+
+		UniProtReactomeEntry uniprot = UniProtReactomeEntry.get(
+			getNextUniProtDBID(), ISOFORM_ACCESSION, DUMMY_UNIPROT_DISPLAY_NAME
+		);
+
+		assertThat(uniprot.getAccession(), equalTo(ISOFORM_ACCESSION));
+	}
+
+	@Test
+	public void incorrectIsoformAccessionThrowsIllegalArgumentException() {
+		final String INCORRECT_ISOFORM_ACCESSION = "P123456-1"; // 7 character parent accession is illegal in UniProt
```

Comment:
Semantics: Would it be better described as Invalid rather than Incorrect?

Reply:
Not sure: http://sparrowflights.blogspot.com/2014/03/incorrect-vs-invalid-vs-illegal.html

Reply:
Looking at the definitions, maybe "Illegal" is the best of the three since it violates UniProt's accession rules.